libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -522,6 +522,7 @@ experimental_table_slice_builder::column_builder::make(
       return std::make_unique<map_column_builder>(pool, std::move(key_builder),
                                                   std::move(value_builder));
     },
+    // TODO: this thing doesn't use a pool, all others do; why?
```

Comment:
I think this should be unified. If we can pass the memory pool to Arrow we should probably do so.

We should also experiment with using `ARROW_DEFAULT_MEMORY_POOL=jemalloc` if we link against jemalloc (it's an optional dependency right now). Can you create a follow-up story for investigation?

Reply:
addressed (pool added for the builder, ticket created)